server/src/main/java/org/elasticsearch/action/admin/indices/rollover/MetadataRolloverService.java

```diff
@@ -70,15 +74,25 @@ public class MetadataRolloverService {
     private static final Logger logger = LogManager.getLogger(MetadataRolloverService.class);
     private static final Pattern INDEX_NAME_PATTERN = Pattern.compile("^.*-\\d+$");
     private static final List<IndexAbstraction.Type> VALID_ROLLOVER_TARGETS = List.of(ALIAS, DATA_STREAM);
-
     public static final Settings HIDDEN_INDEX_SETTINGS = Settings.builder().put(IndexMetadata.SETTING_INDEX_HIDDEN, true).build();
+    public static final Map<AutoShardingType, String> AUTO_SHARDING_METRIC_NAMES = Map.of(
+        AutoShardingType.INCREASE_SHARDS,
+        "es.auto_sharding.increase_shards.total",
+        AutoShardingType.DECREASE_SHARDS,
+        "es.auto_sharding.decrease_shards.total",
+        AutoShardingType.COOLDOWN_PREVENTED_INCREASE,
+        "es.auto_sharding.cooldown_prevented_increase.total",
+        AutoShardingType.COOLDOWN_PREVENTED_DECREASE,
+        "es.auto_sharding.cooldown_prevented_decrease.total"
```

Comment:
The metric names are a function of the action type and could have been computed rather than written as literals. I think it's worth putting literals so the metric names are searchable in the codebase, but I'm open to changing if y'all think otherwise.

Reply:
++